src/cli/cli.cpp

```diff
@@ -2554,6 +2554,51 @@ otError Interpreter::ProcessLinkMetricsProbe(Arg aArgs[])
 }
 #endif // OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
 
+#if OPENTHREAD_CONFIG_TMF_ANYCAST_LOCATOR_ENABLE
+
+otError Interpreter::ProcessLocate(Arg aArgs[])
+{
+    otError      error = OT_ERROR_INVALID_ARGS;
+    otIp6Address anycastAddress;
+
+    if (aArgs[0].IsEmpty())
+    {
+        OutputLine(otThreadIsAnycastLocateInProgress(mInstance) ? "In Progress" : "Idle");
+        ExitNow(error = OT_ERROR_NONE);
+    }
+
+    SuccessOrExit(error = aArgs[0].ParseAsIp6Address(anycastAddress));
+    SuccessOrExit(error = otThreadLocateAnycastDestination(mInstance, &anycastAddress, HandleLocateResult, this));
+
+exit:
+    return error;
+}
+
+void Interpreter::HandleLocateResult(void *              aContext,
+                                     otError             aError,
+                                     const otIp6Address *aMeshLocalAddress,
+                                     uint16_t            aRloc16)
+{
+    static_cast<Interpreter *>(aContext)->HandleLocateResult(aError, aMeshLocalAddress, aRloc16);
+}
+
+void Interpreter::HandleLocateResult(otError aError, const otIp6Address *aMeshLocalAddress, uint16_t aRloc16)
+{
+    OutputFormat("Locate result ");
+
+    if (aError == OT_ERROR_NONE)
+    {
+        OutputIp6Address(*aMeshLocalAddress);
+        OutputLine(" RLOC16: 0x%04x", aRloc16);
+    }
+    else
+    {
+        OutputLine("- Error:%s", otThreadErrorToString(aError));
+    }
+}
```

Comment:
```suggestion

        OutputResult(aError);
}
```

Reply:
We had it as blocking earlier (returning `kErrorPending` and then doing `OutputResult()` from callback). But in the case there is no response, the CoAP timeout is rather longe (up to minutes), and CLI will be blocked during all of this. So I changed it to be non-blocking.

Reply:
Can we use a shorter timeout?

I think the current implementation won't work with `ot-ctl`:
```
# ot-ctl locate fdde:ad00:beef:0:0:ff:fe00:fc00
Done
#
```

Reply:
We can use the newly added CLI `mTimer` for this (to wait for a shorter time ~2second in the CLI) and then make the command be blocking again.  I will change this in next push (with this we can use also change the output format (make it shorter and just output the RLOC16)).

The core implementation uses the default `Coap` with its setting (to keep the code simpler and also has smaller code size). So would be good to keep the core implementation as is.